lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -350,11 +368,8 @@ def main():
 
         # If VPC configuration is desired
         if vpc_subnet_ids or vpc_security_group_ids:
-            if len(vpc_subnet_ids) < 1:
-                module.fail_json(msg='At least 1 subnet is required')
-
-            if len(vpc_security_group_ids) < 1:
-                module.fail_json(msg='At least 1 security group is required')
+            if vpc_subnet_ids is None or len(vpc_subnet_ids) == 0 or vpc_security_group_ids is None or len(vpc_security_group_ids) == 0:
```

Comment:
Easier to read is `if not vpc_subnet_ids or not vpc_security_group_ids:` because None and an empty list will evaluate to false.